src/generator/generator_visitor.ts

```diff
@@ -268,27 +269,15 @@ class GeneratorVisitor extends Visitor {
 
     protected visitMemberExpression(node: MemberExpression) {
         if (node.computed) {
-            super.visitMemberExpression(node);
-
-            const index = this.popExpression();
-            const arrayPointer = this.popExpression();
-            const address = this.module.i32.add(arrayPointer, this.module.i32.mul(index, this.module.i32.const(4)));
-
-            // The offset can not be used, because the memberaccess value can also be a mathematical term or a variable
-            this.expressions.push(this.module.i32.load(0, 4, address));
-        } else {
-            if (isIdentifier(node.property)) {
-                if (node.property.name === 'length') {
-                    this.visit(node.object);
-
-                    const address = this.module.i32.sub(this.popExpression(), this.module.i32.const(4));
-                    this.expressions.push(this.module.i32.load(0, 4, address));
-                } else {
-                    throw new Error(`Unknown property ${node.property}`);
-                }
+            this.getArrayElement(node);
+        } else if (isIdentifier(node.property)) {
+            if (node.property.name === 'length') {
+                this.getArrayLength(node);
             } else {
-                throw new Error('property of static member expression was not an identifier');
+                throw new Error(`Unknown property ${node.property}`);
             }
+        } else {
+            throw new Error('The member access was neither an array access nor was it a property as an identifier');
```

Comment:
```suggestion
            throw new Error('The member access was neither a computer nor a static member expression');
```

By the way, this branch will never be taken because the two branches before already cover all cases. (See [here](https://github.com/babel/babel/blob/master/packages/babel-parser/ast/spec.md#memberexpression))

Reply:
I remove it entirely then. There is no need for code that can never be reached.